src/lib/optimizer/strategy/between_composition_rule.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "abstract_rule.hpp"
+#include "expression/expression_functional.hpp"
+
+#include "types.hpp"
+
+namespace opossum {
+
+class AbstractLQPNode;
+class PredicateNode;
+
+/**
+ * The ColumnBoundaryType defines whether a value represents a boundary for a column or not
+ * and if it is a boundary it also defines which kind of boundary it is including the inclusive and exclusive property.
+**/
+enum ColumnBoundaryType {
+  None,
+  LowerBoundaryInclusive,
+  LowerBoundaryExclusive,
+  UpperBoundaryInclusive,
+  UpperBoundaryExclusive,
+};
+
+/**
+ * A column boundary is a normalized format for further computation
+ * that allows us to store a column and a value expression of a predicate node. The value represents a boundary
+ * for the column, if the ColumnBoundaryType does not equal None.
+ * The PredicateNode has to be stored so it can be removed if a substitution of two ColumnBoundaries is possible.
+**/
+struct ColumnBoundary {
+  std::shared_ptr<PredicateNode> node;
+  std::shared_ptr<LQPColumnExpression> column_expression;
+  std::shared_ptr<ValueExpression> value_expression;
+  ColumnBoundaryType type;
+};
+
+/**
+ * The BetweenCompositionRule searches for a chain of predicate nodes and substitutes BinaryPredicateConditions
```

Comment:
pls introduce your rule with an example like "Rewrite a>=5 AND a<=10 to a BETWEEN 5 AND 10". Makes it far easier to get an idea of what it does